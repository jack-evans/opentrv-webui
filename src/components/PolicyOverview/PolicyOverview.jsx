import React, { Component } from 'react'

class PolicyOverview extends Component {
  render () {
    return (
      <div>
        Policy Overview goes here
      </div>
    )
  }
}

export default PolicyOverview
