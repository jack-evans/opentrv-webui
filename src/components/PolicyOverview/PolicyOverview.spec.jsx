import React from 'react'
import PolicyOverview from './PolicyOverview'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('PolicyOverview.jsx', () => {
  it('renders without crashing', () => {
    expect(shallow(<PolicyOverview />)).toMatchSnapshot()
  })
})
