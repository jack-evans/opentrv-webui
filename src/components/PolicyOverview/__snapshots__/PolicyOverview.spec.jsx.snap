// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyOverview.jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PolicyOverview />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="PolicyOverview__search-bar-container"
        >
          <UNDEFINED>
            <Search
              className="PolicyOverview__search-bar"
              id="policy-search"
              labelText="Search"
              onChange={[Function]}
              placeHolderText="Search Devices"
              small={true}
              type="text"
            />
          </UNDEFINED>
        </div>,
        <div
          className="PolicyOverview__content  PolicyOverview__content-loading"
        >
          <div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Loading
                active={true}
                small={true}
                withOverlay={false}
              />
            </div>
            <div>
              Loading. Please Wait...
            </div>
          </div>
        </div>,
      ],
      "className": "PolicyOverview",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <UNDEFINED>
            <Search
              className="PolicyOverview__search-bar"
              id="policy-search"
              labelText="Search"
              onChange={[Function]}
              placeHolderText="Search Devices"
              small={true}
              type="text"
            />
          </UNDEFINED>,
          "className": "PolicyOverview__search-bar-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Search
              className="PolicyOverview__search-bar"
              id="policy-search"
              labelText="Search"
              onChange={[Function]}
              placeHolderText="Search Devices"
              small={true}
              type="text"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "PolicyOverview__search-bar",
              "id": "policy-search",
              "labelText": "Search",
              "onChange": [Function],
              "placeHolderText": "Search Devices",
              "small": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Symbol(react.fragment),
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <div
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Loading
                active={true}
                small={true}
                withOverlay={false}
              />
            </div>
            <div>
              Loading. Please Wait...
            </div>
          </div>,
          "className": "PolicyOverview__content  PolicyOverview__content-loading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Loading
                  active={true}
                  small={true}
                  withOverlay={false}
                />
              </div>,
              <div>
                Loading. Please Wait...
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Loading
                  active={true}
                  small={true}
                  withOverlay={false}
                />,
                "style": Object {
                  "display": "flex",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": true,
                  "small": true,
                  "withOverlay": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading. Please Wait...",
              },
              "ref": null,
              "rendered": "Loading. Please Wait...",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="PolicyOverview__search-bar-container"
          >
            <UNDEFINED>
              <Search
                className="PolicyOverview__search-bar"
                id="policy-search"
                labelText="Search"
                onChange={[Function]}
                placeHolderText="Search Devices"
                small={true}
                type="text"
              />
            </UNDEFINED>
          </div>,
          <div
            className="PolicyOverview__content  PolicyOverview__content-loading"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Loading
                  active={true}
                  small={true}
                  withOverlay={false}
                />
              </div>
              <div>
                Loading. Please Wait...
              </div>
            </div>
          </div>,
        ],
        "className": "PolicyOverview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <UNDEFINED>
              <Search
                className="PolicyOverview__search-bar"
                id="policy-search"
                labelText="Search"
                onChange={[Function]}
                placeHolderText="Search Devices"
                small={true}
                type="text"
              />
            </UNDEFINED>,
            "className": "PolicyOverview__search-bar-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Search
                className="PolicyOverview__search-bar"
                id="policy-search"
                labelText="Search"
                onChange={[Function]}
                placeHolderText="Search Devices"
                small={true}
                type="text"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "PolicyOverview__search-bar",
                "id": "policy-search",
                "labelText": "Search",
                "onChange": [Function],
                "placeHolderText": "Search Devices",
                "small": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <Loading
                  active={true}
                  small={true}
                  withOverlay={false}
                />
              </div>
              <div>
                Loading. Please Wait...
              </div>
            </div>,
            "className": "PolicyOverview__content  PolicyOverview__content-loading",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Loading
                    active={true}
                    small={true}
                    withOverlay={false}
                  />
                </div>,
                <div>
                  Loading. Please Wait...
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Loading
                    active={true}
                    small={true}
                    withOverlay={false}
                  />,
                  "style": Object {
                    "display": "flex",
                    "justifyContent": "center",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": true,
                    "small": true,
                    "withOverlay": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Loading. Please Wait...",
                },
                "ref": null,
                "rendered": "Loading. Please Wait...",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
